torch/lowrank.py

```diff
@@ -0,0 +1,249 @@
+"""Implement various linear algebra algorithms for low rank matrices.
+"""
+
+__all__ = ['svd', 'pca']
+
+import torch
+from . import _linalg_utils as _utils 
+
+
+def get_approximate_basis(A, q, niter=2, M=None):
+    """Return tensor :math:`Q` with q orthonormal columns such that
+    :math:`Q Q^H A` approximates :math:`A`. If :math:`M` is specified,
+    then `Q` is such that :math:`Q Q^H (A - M)` approximates :math:`A
+    - M`.
+
+    .. note:: The implementation is based on the Algorithm 4.4 from
+              Halko et al, 2009.
+
+    .. note:: For an adequate approximation of a k-rank matrix A,
+              where k is not known in advance but could be estimated,
+              the number of Q columns q can be choosen according to
```

Comment:
q, the number of columns of Q?

Reply:
changed to `the number of Q columns, q, can be..`